### PR TITLE
Generalised Pareto distribution

### PR DESCRIPTION
**Pull request by elenaehrlich**
Description of changes: Addition of the Generalised Pareto distribution

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/distribution/test_distribution_inference.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 # First-party imports
 from gluonts.model.common import NPArrayLike
-from gluonts.mx.distribution.box_cox_transform import (
```
Comment (lostella) [inline test/distribution/test_distribution_inference.py:29]: Looks like this shouldn’t be removed? Same later on in this module

File: src/gluonts/mx/distribution/genpareto.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    Generalised Pareto distribution.
+
+    Parameters
+    ----------
+    xi
+        Tensor containing the xi shape parameters, of shape `(*batch_shape, *event_shape)`.
+    beta
+        Tensor containing the beta scale parameters, of shape `(*batch_shape, *event_shape)`.
+    loc
+        Tensor containing the (for-now positive) location parameters, of shape `(*batch_shape, *event_shape)`.
```
Comment (lostella) [inline src/gluonts/mx/distribution/genpareto.py:48]: Why for now positive? It seems like the corresponding `GenParetoOutput` class doesn’t constrain `loc`

File: src/gluonts/mx/distribution/genpareto.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        Returns
+        -------
+        Tuple[Tensor, Tensor, Tensor]:
+            Three squeezed tensors, of shape `(*batch_shape)`: both have entries mapped to the
+            positive orthant.
+        """
+        epsilon = np.finfo(cls._dtype).eps
+        xi = softplus(F, xi) + epsilon
+        beta = softplus(F, beta) + epsilon
+        # loc = softplus(F, loc) + epsilon
```
Comment (lostella) [inline src/gluonts/mx/distribution/genpareto.py:167]: I guess this can just be removed?

**Comment by lostella** [comment]
Looks good to me, thanks @elenaehrlich. One thing that is missing is the “joint test” for `log_prob` and `sample` that is there for the other distributions: you can see some examples here https://github.com/awslabs/gluon-ts/blob/master/test/distribution/test_distribution_inference.py

Essentially this generates a sample using the distribution, and then fits the parameter with SGD minimizing negative loglikelihood. Would something like that work for GenPareto as well?

File: src/gluonts/mx/distribution/genpareto.py (hunk 2 above)
Comment (elenaehrlich) [reply src/gluonts/mx/distribution/genpareto.py]: To your point about including a test for MLE via SGD, I dropped the inclusion of a location-shift parameter in order to permit this test's inclusion. MLE with a location-shift parameter requires Grimshaw's algorithm [https://amstat.tandfonline.com/doi/abs/10.1080/00401706.1993.10485040], and can perhaps be a future commit.

File: src/gluonts/mx/distribution/genpareto.py (hunk 3 above)
Comment (elenaehrlich) [reply src/gluonts/mx/distribution/genpareto.py]: yes, apologies

**Review by lostella — APPROVED** [review]
Looks good, thanks!

**Pull request merged**